### PR TITLE
Update to handle profile json structure for 3.7.x+

### PR DESCRIPTION
**Pull request by ky-ler**
- Added UserBuilds class to handle the new structure (see below). 

spt aki 3.7.x changes the profile json with the addition of equipment builds.

note: I have not implemented the functionality for equipment builds. Also, these changes are likely not compatible for any profile on versions below 3.7.x.

weapon builds are now found in "userbuilds", along with equipment builds, like so:

```
  "userbuilds": {
    "equipmentBuilds": [],
    "weaponBuilds": [
      "Gunsmith 01": {
        "id": "0815071851457610617fa3cc",
        "name": "Gunsmith 01",
        "root": "47b50284089d64afee7da447",
        "items": [
          {
            "_id": "47b50284089d64afee7da447",
            "_tpl": "54491c4f4bdc2db1078b4568",
            "upd": {
              "Repairable": {
                "MaxDurability": 100,
                "Durability": 100
              },
              "FireMode": {
                "FireMode": "single"
              }
            }
          },
          {
            "_id": "b3cbe4a83d2ddb149054ca9e",
            "_tpl": "55d4491a4bdc2d882f8b456e",
            "parentId": "47b50284089d64afee7da447",
            "slotId": "mod_barrel"
          },
          {
            "_id": "235dc26d42612f7ea2d608e5",
            "_tpl": "55d45f484bdc2d972f8b456d",
            "parentId": "47b50284089d64afee7da447",
            "slotId": "mod_handguard"
          },
          {
            "_id": "fbaca4317f4ae7a1fe4df8e8",
            "_tpl": "55d484b44bdc2d1d4e8b456d",
            "parentId": "47b50284089d64afee7da447",
            "slotId": "mod_magazine"
          },
          {
            "_id": "3a384e63e6e7f7e08f49ada0",
            "_tpl": "56083a334bdc2dc8488b4571",
            "parentId": "47b50284089d64afee7da447",
            "slotId": "mod_stock"
          },
          {
            "_id": "7a34617ff1b271c7fdabb7cd",
            "_tpl": "5c0111ab0db834001966914d",
            "parentId": "b3cbe4a83d2ddb149054ca9e",
            "slotId": "mod_muzzle"
          },
          {
            "_id": "9b5d7de07ad70ad557122893",
            "_tpl": "5c1bc7752e221602b1779b34",
            "parentId": "235dc26d42612f7ea2d608e5",
            "slotId": "mod_foregrip"
          },
          {
            "_id": "0ef5cf3660a8b9c6ccd46568",
            "_tpl": "5cc9c20cd7f00c001336c65d",
            "parentId": "235dc26d42612f7ea2d608e5",
            "slotId": "mod_tactical_001",
            "upd": {
              "Light": {
                "IsActive": true,
                "SelectedMode": 0
              }
            }
          },
          {
            "_id": "835bc4b0c6016a0b70ea7414",
            "_tpl": "58272d7f2459774f6311ddfd",
            "parentId": "7a34617ff1b271c7fdabb7cd",
            "slotId": "mod_muzzle"
          }
        ],
        "Parent": "5447b6094bdc2dc3278b4567"
      }
    }
  ]
```

### CODE REVIEW

**Comment by ky-ler** [comment]
You do not have to merge this to master since it will break compatibility for profiles below 3.7.x. You can create another branch and merge it there if you'd like.

File: SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs [hunk 1]
```diff
@@ -47,19 +47,20 @@ public string[] Suits
...
+        [JsonProperty("userbuilds")]
+        public UserBuilds UserBuilds
         {
-            get => weaponBuilds;
+            get => userBuilds;
             set
             {
-                weaponBuilds = value;
+                userBuilds = value;
+                //OnPropertyChanged("EquipmentBuilds");
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:57]: comments not needed

File: SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs [hunk 2]
```diff
@@ -47,19 +47,20 @@ public string[] Suits
...
                 WeaponBuildsChanged();
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:58]: Is property name changed in this function?

File: SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs [hunk 3]
```diff
@@ -197,13 +198,13 @@ bool SetupQuestFromArray(ActiveQuest[] array, QuestType type)
 
         public void RemoveBuild(string key)
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:199]: Need to change function name - "RemoveWeaponBuild"

File: SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs [hunk 4]
```diff
@@ -197,13 +198,13 @@ bool SetupQuestFromArray(ActiveQuest[] array, QuestType type)
...
         {
-            if (WeaponBuilds.Remove(key))
+            if (UserBuilds.WeaponBuilds.Remove(key))
                 WeaponBuildsChanged();
         }
 
         public void RemoveBuilds()
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:205]: Need to change function name - "RemoveWeaponBuilds"

File: SPT-AKI Profile Editor/Core/ProfileClasses/UserBuilds.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        //[JsonProperty("equipmentBuilds")]
+        //public Dictionary<string, WeaponBuild> EquipmentBuilds
+        //{
+        //    get => equipmentBuilds;
+        //    set
+        //    {
+        //        equipmentBuilds = value;
+        //        OnPropertyChanged("EquipmentBuilds");
+        //    }
+        //}
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Core/ProfileClasses/UserBuilds.cs:23]: comments not needed

File: SPT-AKI Profile Editor/Core/ProfileClasses/UserBuilds.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+        [JsonProperty("weaponBuilds")]
+        public Dictionary<string, WeaponBuild> WeaponBuilds
+        {
+            get => weaponBuilds;
+            set
+            {
+                weaponBuilds = value;
+                OnPropertyChanged("WeaponBuilds");
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Core/ProfileClasses/UserBuilds.cs:32]: nameof(WeaponBuilds) preferred

File: SPT-AKI Profile Editor/Views/WeaponBuildsViewModel.cs [hunk 7]
```diff
@@ -54,7 +54,7 @@ private void ExportAllBuilds()
             var (success, path) = _windowsDialogs.FolderBrowserDialog(true);
             if (success)
             {
-                foreach (var build in Profile.WeaponBuilds)
+                foreach (var build in Profile.UserBuilds.WeaponBuilds)
```
Comment (SkiTles55) [inline SPT-AKI Profile Editor/Views/WeaponBuildsViewModel.cs:57]: Need to change function name - "ExportAllWeaponBuilds"

**Comment by SkiTles55** [comment]
Need also apply changes to tests

**Review by SkiTles55 — COMMENTED** [review]
Changes in tests also needed

File: SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs (hunk 2 above)
Comment (ky-ler) [reply SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:58]: > Is property name changed in this function?

in the WeaponBuildsChanged function?

**Comment by ky-ler** [comment]
> I would also prefer to move all of the weapon build functions to the UserBuilds class.

I think this might get messy once the functions for equipment builds are implemented, but I can do if you would still prefer.

File: SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs (hunk 2 above)
Comment (SkiTles55) [reply SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:58]: > in the WeaponBuildsChanged function?

Yes, OnPropertyChanged is called here for few properties, including the old WeaponBuilds

Comment (ky-ler) [reply SPT-AKI Profile Editor/Core/ProfileClasses/Profile.cs:58]: > > in the WeaponBuildsChanged function?
> 
> Yes, OnPropertyChanged is called here for few properties, including the old WeaponBuilds

Oh yeah, sorry. I just updated it.

**Pull request closed without merging**